src/main/java/uk/gov/hmcts/darts/dailylist/service/DailyListProcessorImpl.java

```diff
@@ -0,0 +1,181 @@
+            if (!dailyLists.isEmpty()) {
+                try {
+                    processDailyList(dailyLists.get(0));
+                } catch (JsonProcessingException e) {
+                    throw new DartsApiException(DailyListError.FAILED_TO_PROCESS_DAILYLIST, e);
+                }
+
+            }
+            if (dailyLists.size() > 1) {
+                ignoreOldDailyList(dailyLists.subList(1, dailyLists.size()));
+            }
+        }
+    }
+
+    private void ignoreOldDailyList(List<DailyListEntity> dailyLists) {
+        for (DailyListEntity dailyList : dailyLists) {
+            dailyList.setStatus(String.valueOf(JobStatusType.IGNORED));
+        }
+    }
+
+    @SuppressWarnings("checkstyle:VariableDeclarationUsageDistance")
+    private void processDailyList(DailyListEntity dailyListEntity) throws JsonProcessingException {
+        DailyList dailyList = objectMapper.readValue(dailyListEntity.getContent(), DailyList.class);
+        JobStatusType statusType = JobStatusType.PROCESSED;
+
+        for (CourtList courtList : dailyList.getCourtLists()) {
+
+            String courtHouseName = courtList.getCourtHouse().getCourtHouseName();
+            Optional<CourthouseEntity> foundCourthouse = courthouseRepository.findByCourthouseNameIgnoreCase(
+                    courtHouseName);
+
+            if (foundCourthouse.isPresent()) {
+                List<Sitting> sittings = courtList.getSittings();
+                for (Sitting sitting : sittings) {
+                    List<Hearing> hearings = sitting.getHearings();
+                    for (Hearing dailyListHearing : hearings) {
+
+                        String caseNumber = getCaseNumber(dailyListHearing.getDefendants(), dailyListEntity, dailyListHearing);
+
+                        HearingEntity hearing = retrieveCoreObjectService.retrieveOrCreateHearing(
+                                courtHouseName, String.valueOf(sitting.getCourtRoomNumber()),
+                                caseNumber, dailyListHearing.getHearingDetails().getHearingDate()
+                        );
+
+                        CourtCaseEntity courtCase = hearing.getCourtCase();
+                        addJudges(sitting, hearing);
+                        addDefendants(courtCase, dailyListHearing.getDefendants());
+                        addProsecutions(courtCase, dailyListHearing.getProsecution().getAdvocate().getPersonalDetails().getName());
+                        addDefenders(courtCase, dailyListHearing.getDefendants());
+
+                        courtCase.addHearing(hearing);
+
+                        hearingRepository.saveAndFlush(hearing);
+                        caseRepository.saveAndFlush(courtCase);
+                    }
+                }
+            } else {
+                statusType = JobStatusType.PARTIALLY_PROCESSED;
+                log.error("Unregistered courthouse " + courtHouseName + " daily list entry has not been processed");
```

Comment:
we should maybe give the id of the dailyList we are processing to help 2nd line support track down an issue.

Reply:
ACCEPT